Microsoft.Toolkit/Diagnostics/Generated/ThrowHelper.Collection.tt

```diff
@@ -5,131 +5,125 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
-using System.Runtime.CompilerServices;
 using Microsoft.Toolkit.Extensions;
 
 namespace Microsoft.Toolkit.Diagnostics
 {
     /// <summary>
-    /// Helper methods to efficiently throw exceptions.
+    /// Helper methods to verify conditions when running code.
     /// </summary>
-    public static partial class ThrowHelper
+    public static partial class Guard
     {
+        /// <summary>
+        /// Helper methods to efficiently throw exceptions.
+        /// </summary>
+        private static partial class ThrowHelper
```

Comment:
Why is this being wrapped in a class now? I think that's the main thing I'm confused about in the PR (and with all the additional new ThrowHelper wraps below). Is this a copy of the other ThrowHelpers or something else?

Reply:
All these methods have always been wrapped in a class, it's just that before they were inside the public `ThrowHelper` class, and internal, while now they're all public and inside a new, private `ThrowHelper` class that's nested inside `Guard`. There is no behavioral different, it's just a code reorganization. The rationale for this is to make the code compatible (or rather, to make it better) in relation to #3587. The idea being that if we publish the source-only package and wanted to just copy-paste the `Guard` and `ThrowHelper` classes into consuming projects, they would've been able to also see all those internal helpers, which would've just cluttered their IntelliSense. With this structure instead all those internal helpers would still be invisible outside of the `Guard` class, even if the class was just added as source code directly into your project 🙂